docs: add float & integer documentation on parameter & return value (#9408)
dagger/dagger: docs/current_docs/api/arguments.mdx

HELLO, WORLD
```

## Integer arguments

To pass an integer argument to a Dagger function, add the corresponding flag to the `dagger call` command, followed by the integer value.

> Here is an example of a Dagger function that accepts a integer argument:

Reviewer: ```suggestion
Here is an example of a Dagger function that accepts an integer argument:
```

<Tabs groupId="language">

<TabItem value="Go">
```go file=./snippets/functions/arguments-integer/go/main.go
```
</TabItem>

<TabItem value="Python">
```python file=./snippets/functions/arguments-integer/python/main.py
```
</TabItem>

<TabItem value="TypeScript">
```typescript file=./snippets/functions/arguments-integer/typescript/index.ts
```
</TabItem>

</Tabs>

Here is an example call for this Dagger Function:

```shell
dagger call add-integer --a 1 --b 2
```

The result will look like this:

```shell
3
```

> ## Float arguments

Reviewer: ```suggestion
## Floating-point number arguments
```

:::warning

> Floats are currently only supported in Go and TypeScript SDKs for now.

Reviewer: The information in this section is only applicable to the Go and TypeScript SDKs. Floating-point number arguments are not currently supported in the Go SDKs.

Author: Actually it's now supported in Python so I'll remove this msg :D

:::

> To pass a float argument to a Dagger function, add the corresponding flag to the `dagger call` command, followed by the float value.

Reviewer: ```suggestion
To pass a floating-point number as argument to a Dagger function, add the corresponding flag to the `dagger call` command, followed by the value.
```

> Here is an example of a Dagger function that accepts a float argument:

Reviewer: ```suggestion
Here is an example of a Dagger function that accepts a floating-point number as argument:
```

<Tabs groupId="language">
<TabItem value="Go">
```go file=./snippets/functions/arguments-float/go/main.go
```
</TabItem>
<TabItem value="TypeScript">

:::note
There's no `float` type keyword in TypeScript because the type keyword `number` already supports floating point numbers.

> To declare a `float` argument on the function signature, you need to import `float` from `@dagger.io/dagger` and use it as argument's type.

Reviewer: ```suggestion
To declare a `float` argument on the function signature, import `float` from `@dagger.io/dagger` and use it as an argument's type.
```
